### ISSUE TITLE
AccessibilityInfo.announceForAccessibility does not work from XamlIslands

### ISSUE THREAD

**Issue by rozele** [microsoft/react-native-windows#7245]
## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Copied AccessibilityInfoModule approach into sample XamlIslands app (create TextBlock, use to create automation peer)
2. Turn on narrator, trigger logic
3. Narrator does not announce the raised event.

## Expected Results

Narrator should speak the text parameter supplied to the announceForAccessibility method. I suspect the automation peer does not work as expected from an element that is not attached to the XamlRoot in XamlIslands.

The previous implementation (added by @acoates-ms) grabbed the XamlRoot from the ReactContext property bag or Window.Content(), but neither of these are guaranteed to create an automation peer.

## Snack, code example, screenshot, or link to a repository:

Here's the repro of the `announceForAccessibility` approach in a standalone XAML Islands app:
https://github.com/rozele/Xaml-Islands-Samples/tree/accessibilityExample

**Comment by rozele** [comment]
cc @chiaramooney

**Comment by asklar** [comment]
@rozele: @chiaramooney changed the object that it's announced on to be a textbox so this should work, let us know if that's not the case.

**Comment by rozele** [comment]
@asklar - I applied this approach in a standalone XAML Islands app (w/o RNW) and it does not work:

**Comment by asklar** [comment]
Just checked in the XAML source; we do need the object to be part of the tree leading up to a XamlRoot, so it cannot be a just-newed up control at least for xaml islands scenarios
